File: app/src/main/java/com/kiwiandroiddev/sc2buildassistant/feature/player/domain/BuildPlayerEventListener.java
package com.kiwiandroiddev.sc2buildassistant.feature.player.domain;


import com.kiwiandroiddev.sc2buildassistant.domain.entity.BuildItem;

/** Interface used by BuildPlayer to notify high-level code of playback events */
public interface BuildPlayerEventListener {
	void onBuildThisNow(BuildItem item, int position);
	void onBuildPlay();
	void onBuildPaused();
	void onBuildStopped();
	void onBuildResumed();
	void onBuildFinished();
	void onIterate(long newGameTimeMs);
}

File: app/src/main/java/com/kiwiandroiddev/sc2buildassistant/feature/player/domain/BuildPlayerEventListener.kt
package com.kiwiandroiddev.sc2buildassistant.feature.player.domain


import com.kiwiandroiddev.sc2buildassistant.domain.entity.BuildItem

/** Interface used by BuildPlayer to notify high-level code of playback events  */
interface BuildPlayerEventListener {
    fun onBuildThisNow(item: BuildItem, position: Int)
    fun onBuildPlay()
    fun onBuildPaused()
    fun onBuildStopped()
    fun onBuildResumed()
    fun onBuildFinished()
    fun onBuildItemsChanged(newBuildItems: List<BuildItem>)
    fun onIterate(newGameTimeMs: Long)
}
